sys/hash_string/hash_string.c

```diff
@@ -26,7 +26,7 @@ unsigned long hash_string(unsigned char *str)
     unsigned long hash = 5381;
     int c;
 
-    while ((c = *str++)) {
```

Comment:
The extra parens are intentionally here, because using `=` in a condition is almost always a typo.
I'd prefer `for (char c; (c = *str) != 0; ++str)`, though.